### ISSUE TITLE
no-useless-escape fail checking \[a\] 

### ISSUE THREAD

**Issue by popomore** [eslint/eslint#7507]
**Tell us about your environment**

* **ESLint Version:** 3.9.1
* **Node Version:** 6.7.0
* **npm Version:** 3.10.3

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

You can find it in https://github.com/eggjs/eslint-config-egg

**What did you do? Please include the actual source code causing the issue.**

```js
console.log(/^\[a\]$/.test('[a]'));
```

**What did you expect to happen?**

It will return true, and eslint won't fail

**What actually happened? Please include the actual, raw output from ESLint.**

eslint failed with `error  Unnecessary escape character: \]  no-useless-escape`

**Comment by platinumazure** [comment]
Since it's not in a character class, `\]` is a useless escape. (The `]` character only has special meaning when inside a character class, since it has to close the character class.)

As far as I can tell, the rule is behaving as designed (it was a bug that the rule wasn't flagging that escape).

**Comment by popomore** [comment]
So I should remove the latter `\`, and reserve the former, like `console.log(/^\[a]$/.test('[a]'));`, am I right?

**Comment by platinumazure** [comment]
@popomore Yes, exactly.

**Comment by popomore** [comment]
@platinumazure Thx
